pkg/operator/controller/ingress/controller.go

```diff
@@ -413,14 +413,25 @@ func (r *reconciler) admit(current *operatorv1.IngressController, ingressConfig
 
 // needsReadmission returns a Boolean value indicating whether the given
 // ingresscontroller needs to be re-admitted.  Re-admission is necessary in
-// order to revalidate mutable fields that are subject to admission checks.  The
-// determination whether re-admission is needed is based on the
+// order to revalidate mutable fields that are subject to admission checks.  To
+// determine whether re-admission is needed, this function compares the
 // ingresscontroller's current generation and the observed generation recorded
-// in its status.
+// in its status.  This function also checks whether status is missing required
+// fields, which could indicate that someone or something else other than this
+// operator has tampered with status.
 func needsReadmission(ic *operatorv1.IngressController) bool {
 	if ic.Generation != ic.Status.ObservedGeneration {
 		return true
 	}
+	if ic.Status.EndpointPublishingStrategy == nil {
```

Comment:
@Miciah I think this solves my concerns (I will re-check the overall PR, I am going commit by commit now), but if we re-admit and during readmission we set the status, I think this is fine

Reply:
amazing, thanks!